Rebrand of Azure Active Directory to Microsoft Entra ID (#5054)
dbt-labs/docs.getdbt.com: website/docs/docs/cloud/git/setup-azure.md

---
title: "Set up Azure DevOps"
id: "setup-azure"
description: "You can set up your Azure DevOps by creating a Microsoft Entra ID app and adding it to dbt Cloud."
sidebar_label: "Set up Azure DevOps"
---

<Snippet path="available-enterprise-tier-only" />

## Overview

To use our native integration with Azure DevOps in dbt Cloud, an account admin needs to set up a Microsoft Entra ID app. We recommend setting up a separate [Microsoft Entra ID application than used for SSO](/docs/cloud/manage-access/set-up-sso-microsoft-entra-id).

1. [Register a Microsoft Entra ID app](#register-a-microsoft-entra-id-app).
2. [Add permissions to your new app](#add-permissions-to-your-new-app).
3. [Add another redirect URI](#add-another-redirect-uri).
4. [Connect Azure DevOps to your new app](#connect-azure-devops-to-your-new-app).
5. [Add your Microsoft Entra ID app to dbt Cloud](#add-your-microsoft-entra-id-app-to-dbt-cloud).

Once the Microsoft Entra ID app is added to dbt Cloud, an account admin must also connect a service user via OAuth, which will be used to power headless actions in dbt Cloud such as deployment runs and CI.

> 1. [Connecting a Service User](#connecting-a-service-user).

Reviewer: fixing link
```suggestion
1. [Connecting a Service User](#connect-a-service-user).
```

Reviewer: hey @nghi-ly , i'll reopen this again and commit there sugg as it's still not going to work since the h2 is 'connect'

Author: weird, i remember clicking "Commit suggestion" for this. doing it again...

Author: i went to update and it's already committed 🤷‍♀️ resolving ... again

![Screenshot 2024-03-21 at 9 10 31 AM](https://github.com/dbt-labs/docs.getdbt.com/assets/107218380/3a7fc38d-7b88-4992-a3f6-1c740bdd132b)

## Register a Microsoft Entra ID app

1. Sign into your Azure portal and click **Microsoft Entra ID** under Azure services.
2. Select **App registrations** in the left panel.
3. Select **New registration**. The form for creating a new Microsoft Entra ID app opens.
4. Provide a name for your app. We recommend using, "dbt Labs Azure DevOps App".
5. Select **Accounts in any organizational directory (Any Microsoft Entra ID directory - Multitenant)** as the Supported Account Types.
Many customers ask why they need to select Multitenant instead of Single tenant, and they frequently get this step wrong. Microsoft considers Azure DevOps (formerly called Visual Studio) and Microsoft Entra ID as separate tenants, and in order for this Microsoft Entra ID application to work properly, you must select Multitenant.
6. Add a redirect URI by selecting **Web** and, in the field, entering `https://YOUR_ACCESS_URL/complete/microsoft_entra_id`, replacing `YOUR_ACCESS_URL` with the [appropriate Access URL](/docs/cloud/about-cloud/access-regions-ip-addresses) for your region and plan.
7. Click **Register**.

> <Lightbox src="/img/docs/dbt-cloud/connecting-azure-devops/ADnavigation.gif" title="Navigating to the Azure AD app registrations"/>

Reviewer: should it be ms entra id?
```suggestion
<Lightbox src="/img/docs/dbt-cloud/connecting-azure-devops/ADnavigation.gif" title="Navigating to the Microsoft Entra app registrations"/>
```

Author: logging for posterity: chatted with docs team about the two options (mismatched caption/image vs still referencing AAD). as a group, we leaned towards first option

> To use our native integration with Azure DevOps in dbt Cloud, an account admin needs to set up an Microsoft Entra ID app. We recommend setting up a separate [Entra ID application than used for SSO](/docs/cloud/manage-access/set-up-sso-azure-active-directory).

Reviewer 2: That's my bad. I edited this page 
```suggestion
To use our native integration with Azure DevOps in dbt Cloud, an account admin needs to set up an Microsoft Entra ID app. We recommend setting up a separate [Entra ID application than used for SSO](/docs/cloud/manage-access/set-up-sso-microsoft-entra-id).
```

> 

Reviewer: above this, i think there's an image that references AD - should that change to ms entra id?

![Screenshot 2024-03-21 at 12 44 56](https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/3994901a-1f09-4097-9b1d-90720bc40f1c)

Author: got it -- nice catch!

## Connect Azure DevOps to your new app
If you have already connected your Azure DevOps account to Microsoft Entra ID, then you can proceed to [Connecting a service user](#connecting-a-service-user). However, if you're just getting set up, connect Azure DevOps to the Microsoft Entra ID App you just created:

1. From your Azure DevOps account, select **Organization settings** in the bottom left.
2. Navigate to Microsoft Entra ID.
3. Click **Connect directory**.
4. Select the directory you want to connect.
5. Click **Connect**.

> <Lightbox src="/img/docs/dbt-cloud/connecting-azure-devops/connect AD to Azure DevOps.gif" title="Connecting Azure DevOps and Active Directory"/>

Reviewer: shouldit be microsoft entra id?
```suggestion
<Lightbox src="/img/docs/dbt-cloud/connecting-azure-devops/connect AD to Azure DevOps.gif" title="Connecting Azure DevOps and Microsoft Entra ID"/>
```

Author: logging for posterity: chatted with docs team about the two options (mismatched caption/image vs still referencing AAD). as a group, we leaned towards first option

## Add your Microsoft Entra ID app to dbt Cloud

Once you connect your Microsoft Entra ID app and Azure DevOps, you need to provide dbt Cloud information about the app:

1. Navigate to your account settings in dbt Cloud.
2. Select **Integrations**.
3. Scroll to the Azure DevOps section.
4. Complete the form:
    - **Azure DevOps Organization:** Must match the name of your Azure DevOps organization exactly. Do not include the `dev.azure.com/` prefix in this field. ✅ Use `my-devops-org` ❌ Avoid `dev.azure.com/my-devops-org`
    - **Application (client) ID:** Found in the Microsoft Entra ID App.
    - **Client Secrets:** You need to first create a secret in the Microsoft Entra ID App under **Client credentials**. Make sure to copy the **Value** field in the Microsoft Entra ID App and paste it in the **Client Secret** field in dbt Cloud. You are responsible for the Microsoft Entra ID app secret expiration and rotation.
    - **Directory(tenant) ID:** Found in the Microsoft Entra ID App.

>         <Lightbox src="/img/docs/dbt-cloud/connecting-azure-devops/AzureDevopsAppdbtCloud.gif" title="Adding an Active Directory App to dbt Cloud"/>

Reviewer: ```suggestion
<Lightbox src="/img/docs/dbt-cloud/connecting-azure-devops/AzureDevopsAppdbtCloud.gif" title="Adding a Microsoft Entra app to dbt Cloud"/>
```

Author: nice catch! 

looked at the MS site, found https://learn.microsoft.com/en-us/entra/identity/enterprise-apps/overview-application-gallery , will mimic usage of "MS Entra app" (sans ID)
